### PR TITLE
[Merged by Bors] - Fix clippy warnings

### LINKED ISSUE #1977: Fix clippy warnings from version 1.57.0

### LINKED ISSUE #1978: Fix --update-context flag in `fluvio cluster start --k8` command
Right now it is being ignored

### PR DESCRIPTION
**Pull request by morenol**
Closes #1977 
Closes  #1978 (by removing the flag)

### CODE REVIEW

File: crates/fluvio-cluster/src/start/k8.rs [hunk 1]
```diff
@@ -579,8 +580,6 @@ pub struct ClusterInstaller {
     config: ClusterConfig,
     /// Shared Kubernetes client for install
     kube_client: SharedK8Client,
-    /// Helm client for performing installs
-    helm_client: HelmClient,
```
Comment (morenol) [inline crates/fluvio-cluster/src/start/k8.rs:583]: This is not being used,  do we have to use this?

File: crates/fluvio-cluster/src/start/local.rs [hunk 2]
```diff
@@ -149,25 +148,6 @@ pub struct LocalConfig {
...
-    /// # use fluvio_cluster::{ClusterError, LocalConfigBuilder};
-    /// # fn example(builder: &mut LocalConfigBuilder) -> Result<(), ClusterError> {
-    /// let config = builder
-    ///     .install_sys(false)
-    ///     .build()?;
-    /// # Ok(())
-    /// # }
-    /// ```
-    #[builder(default = "true")]
-    install_sys: bool,
```
Comment (morenol) [inline crates/fluvio-cluster/src/start/local.rs:170]: I think that this is not needed for `--local`. It is not being used

File: crates/fluvio-sc/src/controllers/topics/controller.rs [hunk 3]
```diff
@@ -20,7 +20,7 @@ use super::reducer::TopicReducer;
 pub struct TopicController {
     topics: StoreContext<TopicSpec>,
     partitions: StoreContext<PartitionSpec>,
-    spus: StoreContext<SpuSpec>,
+    _spus: StoreContext<SpuSpec>,
```
Comment (morenol) [inline crates/fluvio-sc/src/controllers/topics/controller.rs:23]: not being used, should we remove this?

File: crates/fluvio-spu/src/replication/follower/controller.rs [hunk 4]
```diff
@@ -424,14 +424,14 @@ mod inner {
     /// Used to communicate changes to Group Controller
     #[derive(Debug)]
     pub struct GroupNotification {
-        spu: SpuId,
+        _spu: SpuId,
```
Comment (morenol) [inline crates/fluvio-spu/src/replication/follower/controller.rs:427]: should we remove this?

File: crates/fluvio-cluster/src/start/k8.rs (hunk 1 above)
Comment (sehz) [reply crates/fluvio-cluster/src/start/k8.rs:583]: let's remove unused

File: crates/fluvio-cluster/src/start/local.rs (hunk 2 above)
Comment (sehz) [reply crates/fluvio-cluster/src/start/local.rs:170]: k

File: crates/fluvio-sc/src/controllers/topics/controller.rs (hunk 3 above)
Comment (sehz) [reply crates/fluvio-sc/src/controllers/topics/controller.rs:23]: k

File: crates/fluvio-spu/src/replication/follower/controller.rs (hunk 4 above)
Comment (sehz) [reply crates/fluvio-spu/src/replication/follower/controller.rs:427]: k

**Review by sehz — APPROVED** [review]
LGTG

**Pull request closed without merging**